assets/arc-0060/arc60wallet.api.ts

```diff
@@ -0,0 +1,324 @@
+import {
+    crypto_core_ed25519_scalar_add,
+    crypto_core_ed25519_scalar_mul,
+    crypto_core_ed25519_scalar_reduce,
+    crypto_hash_sha512,
+    crypto_scalarmult_ed25519_base_noclamp,
+    ready
+} from 'libsodium-wrappers-sumo'
+
+import * as crypto from 'crypto'
+import { readFileSync } from 'fs';
+import path from 'path';
+import Ajv, { JSONSchemaType } from 'ajv';
+import sha512, { sha512_256 } from 'js-sha512'
+import axios, { AxiosResponse } from 'axios'
+import * as util from 'util'
+
+export interface ARC47LsigTemplateRequest {
+    LogicSignatureDescription: {  // sha512_256 hash of this object is the template hash
+        name: string,
+        description: string,
+        program: string, // base64 encoded TEAL program
+        variables: Array<{
+            variable: string,
+            name: string,
+            type: string,
+            description: string
+        }>
+    },
+    hash: string, // hash of the program, AFTER values replaced and compiled
+    values: { // to replace on the program
+        [key: string]: any
+    }
+}
+
+export interface HDWalletMetadata {
+    /**
+    * HD Wallet purpose value. First derivation path level. 
+    * Hardened derivation is used.
+    */
+    purpose: number,
+
+    /**
+    * HD Wallet coin type value. Second derivation path level.
+    * Hardened derivation is used.
+    */
+    coinType: number,
+
+    /**
+    * HD Wallet account number. Third derivation path level.
+    * Hardened derivation is used.
+    */
+    account: number,
+
+    /**
+    * HD Wallet change value. Fourth derivation path level.
+    * Soft derivation is used.
+    */
+    change: number,
+
+    /**
+    * HD Wallet address index value. Fifth derivation path level.
+    * Soft derivation is used.
+    */
+    addrIdx: number,
+}
+
+// StdSigData type
+export interface StdSigData {
+    data: string;
+    signer: Uint8Array;
+    hdPath?: HDWalletMetadata
+}
+
+// ScopeType type
+export enum ScopeType {
+    UNKNOWN = -1,
+    CHALLENGE32 = 0,
+    MX_RANDOM = 1,
```

Comment:
What's the rationale of random bytes now? We have had very long discussions on "random is not good".

Reply:
MX_RANDOM is for backwards compatibility purposes as there are some private API's out there that use it. The MX prefix makes it safe to sign as i can't be anything else than a arbitrary message. Still open to remove / tweak it though.

We can test the implementation is an adversarial scenario if someone finds one.